python/plugins/processing/algs/qgis/ui/FieldsMappingPanel.py

```diff
@@ -87,20 +87,25 @@ def testExpression(self, row):
         self._errors[row] = None
         field = self._mapping[row]
         expression = QgsExpression(field['expression'])
+        context = self.contextGenerator().createExpressionContext()
+        expression.prepare(context)
```

Comment:
I would move that down just before the loop (check for parser errors before preparing)